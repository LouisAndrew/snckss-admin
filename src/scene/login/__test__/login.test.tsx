import React, { Suspense } from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { FirebaseAppProvider } from 'reactfire'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import config from '../../../lib/firebase-config'
import Login from '..'

// TODO: how to test login auth?
describe('Login component', () => {
        afterEach(cleanup)
        const el: React.ReactElement = (
                <FirebaseAppProvider firebaseConfig={config}>
                        <Suspense fallback={<></>}>
                                <Login />
                        </Suspense>
                </FirebaseAppProvider>
        )

        it('renders without crashing', () => {
                const div = document.createElement('div')
                ReactDOM.render(el, div)
        })

        // it('renders correctly', () => {

        //         const { getByTestId } = render()
        // })

        it('matches snapshot', () => {
                const tree = renderer.create(el).toJSON()
                expect(tree).toMatchSnapshot()
        })
})
