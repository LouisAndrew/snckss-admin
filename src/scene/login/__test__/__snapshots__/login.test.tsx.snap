// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component matches snapshot 1`] = `
<div
  className="wrapper"
>
  <div
    className="login"
  >
    <h1>
      Snckss Admin Panel
    </h1>
    <h3>
      Please login to continue
    </h3>
    <button
      aria-label={null}
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Login with google
    </button>
  </div>
</div>
`;
